pkg/archive/archive_test.go

```diff
@@ -724,6 +726,65 @@ func TestTarUntar(t *testing.T) {
 	}
 }
 
+func TestTarAppenderChownOption(t *testing.T) {
+	origin, err := ioutil.TempDir("", "docker-test-tar-chown-opt")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(origin)
+	if err := ioutil.WriteFile(filepath.Join(origin, "1"), []byte("hello world"), 0700); err != nil {
+		t.Fatal(err)
+	}
+
+	pipeReader, pipeWriter := io.Pipe()
+	errCh := make(chan error)
+	go func() {
+		ta := newTarAppender(
+			&idtools.IDMappings{},
+			pipeWriter,
+			&idtools.IDPair{UID: 1337, GID: 42},
+		)
+
+		defer func() {
+			if err := ta.TarWriter.Close(); err != nil {
+				errCh <- err
+			}
+			if err := pipeWriter.Close(); err != nil {
+				errCh <- err
+			}
+		}()
+		defer pools.BufioWriter32KPool.Put(ta.Buffer)
+
+		if err := ta.addTarFile(filepath.Join(origin, "1"), "1"); err != nil {
```

Comment:
Writes to the pipe will block if there is nothing reading it. `bytes.Buffer` would probably be acceptable for this test instead of a pipe to keep it simple. 
As `tarAppender` is private structure, it is not so important to test it specifically. I'd rather call `TarWithOptions` with the new option.